client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -41,9 +41,12 @@ object Retry {
             case Some(duration) =>
               // info instead of error(e), because e is not discarded
               logger.info(
-                s"Request $req threw an exception on attempt #$attempts attempts. Giving up.")
+                s"Request $req threw an exception on attempt #$attempts with message: ${e.getMessage}. Retrying after $duration.")
```

Comment:
I think we should log the traces here.  It could get noisy, but if we don't, we'll never see the discarded exception.  (Line 42 should read "because we're trying again" and not "because e is discarded")

Reply:
Hi @rossabaker - I don't entirely follow. Could you please say more as to what changes you're recommending? Thanks

Reply:
`logger.info(e)(s"Request threw an exception on attempt #$attempts. Retrying after $duration")`

Without that, all we get is `${e.getMessage}`, which very often is `null`?  (What?  Where?  Who knows!)  When we pass `(e)` to `logger.info(e)`, we can diagnose the intermittent failures that resolve themselves.

Reply:
Thanks for the explanation! I added this change in https://github.com/http4s/http4s/pull/2048/commits/c794efa52f2a70bf97bb9c9111bf8fe291d45707.